cassiopeia/solver/dissimilarity_functions.py

```diff
@@ -195,17 +167,17 @@ def weighted_hamming_similarity(
 
         if s1[i] == s2[i]:
             if s1[i] != 0:
-                if w:
-                    d += 2 * w[i][s1[i]]
+                if weights:
+                    d += 2 * weights[i][s1[i]]
                 else:
                     d += 2
             else:
-                if not w:
+                if not weights:
                     d += 1
         else:
             if s1[i] != 0 and s2[i] != 0:
-                if w:
-                    d -= w[i][s1[i]] - w[i][s2[i]]
+                if weights:
+                    d -= weights[i][s1[i]] + weights[i][s2[i]]
```

Comment:
same as above for the hamming_dissimilarity -- we should only be tallying similarities here, I think.